src/config/rcxml.c

```diff
@@ -748,6 +833,24 @@ xml_tree_walk(xmlNode *node)
 			in_window_rules = false;
 			continue;
 		}
+		if (!strcasecmp((char *)n->name, "query")) {
```

Comment:
Considering protecting against nested if-actions and foreach-actions.
For example if in an `<else>` branch, should we accept another `<query>` (or `if`/`foreach` action).

Reply:
I've added the logic to ignore child If and ForEach. As for query, there is no easy way with current parser to prevent such things.